llvm/lib/Target/AArch64/AArch64SpeculationHardening.cpp

```diff
@@ -299,7 +299,7 @@ bool AArch64SpeculationHardening::instrumentControlFlow(
     if (I == MBB.begin())
```

Comment:
The `I == MBB.begin()` is not needed. We could clean this up further by just calling `RS.backward(I)` unconditionally. I left it in just in case there was some performance reason for having a special case for start-of-block.